.github/workflows/ui-tests.yml

```diff
@@ -3,17 +3,17 @@ name: UI Tests
 on:
   workflow_dispatch:
   schedule:
-    - cron: '0 1 * * *'
+    - cron: '0 1 * * 0-4'
   pull_request:
+    branches: [ master ]
     types:
       - opened
       - edited
       - synchronize
-    branches:
-      - 'release/next'
 
 jobs:
   build-and-upload-to-firebase-and-browserstack:
+    if: startsWith(github.head_ref, 'release/') || github.head_ref == 'master'
```

Comment:
The trigger matches against the `base` not the head.
I believe this setup gets us:

Run on any PR pointed to master, only if the head is `release/**`
The check for `|| github.head_ref == 'master'` is to allow nightly runs to pass, and for manual runs (which must be triggered on master in this case)

Thoughts @HarryPrimer @semirp

Reply:
`- run: echo "BRANCH_NAME=${{ github.event.pull_request && github.head_ref || github.ref_name }}" >> $GITHUB_ENV`

This is a good way to get the branch name in all cases, then it could work just to compare current branch new and what we want to execute.

I believe it could work if we only change `github.ref_name == 'master'`

Reply:
Good shout. But I see another problem here. It will run on any PR pointing to `master` currently, which is not what I want.

I want to satisfy all of these scenarios:
- Run on all pull requests with the head of `release/**`
- Allow nightly runs to run on `master`
- Allow manual runs to run on `master`

🤔

Reply:
Using `- run: echo "BRANCH_NAME=${{ github.event.pull_request && github.head_ref || github.ref_name }}"`, I believe we could use 

`if: $BRANCH_NAME == "release/next" || $BRANCH_NAME == "master"`

I think that is correct, do you think it is cleaner than the above? @semirp

Reply:
Yeah that would work, sorry I wanted to say that it could work even if we just change it to:
` startsWith(github.head_ref, 'release/') || github.ref_name == 'master'`

but I believe targeting branches as you described could be even simpler